docs/user-guide/model.rst

```diff
@@ -0,0 +1,173 @@
+
+*****
+Table
+*****
+
+A `table <../dev/internals/felis.datamodel.Schema.html#felis.datamodel.Table>`__ represents a collection of
+columns along with their indexes and constraints and has these attributes:
+
+:``name``: The name of this table. This is the name that will be used to create the table in the database.
+:``@id``: A unique identifier for this table.
+:``description``: A textual description of this table.
+:``columns``: The list of columns in the table.
+:``primaryKey``: A list of column names that make up the primary key for this table.
+:``constraints``: The list of constraints for the table. The most typical type is a foreign key constraint.
+:``indexes``: The list of indexes for the table.
+
+A table may also have the following optional attributes:
+
+:``tap:table_index``: The index of this table in a TAP schema. This is used to order the tables in the schema.
+:``mysql:charset``: The `MySQL character set <https://dev.mysql.com/doc/refman/8.4/en/charset-charsets.html>`__ for the table.
+:``mysql:engine``: The `MySQL storage engine <https://dev.mysql.com/doc/refman/8.4/en/storage-engines.html>`__ for the table.
+
+******
+Column
+******
+
+A `column <../dev/internals/felis.datamodel.Schema.html#felis.datamodel.Column>`__ represents a single field
+in a table.
+A column has a name and a data type and may have additional metadata like units of measurement.
+
+:``name``: The name of this column. This is the name that will be used to create the column in the database.
+:``@id``: A unique identifier for this column.
+:``description``: A textual description of this column.
+:``datatype``: The data type of this column. Valid datatypes are defined by the `Felis type system <../dev/internals.html#module-felis.types>`__.
+:``autoincrement``: A boolean flag indicating whether this column is auto-incrementing.
+:``length``: The length of the column. This is only applicable to certain data types.
+:``nullable``: A boolean flag indicating whether this column can be null.
+:``precision``: The precision of the column. This is current only applicable to columns with the ``timestamp`` datatype.
+:``value``: The default value for this column.
+
+A column may also have the following optional properties:
+
+:``fits:tunit``: The `FITS TUNIT <https://fits.gsfc.nasa.gov/standard30/fits_standard30aa.pdf>`__ unit for this column [1]_.
+:``ivoa:ucd``: The `IVOA UCD <http://www.ivoa.net/documents/latest/UCD.html>`__ for this column [1]_.
+:``ivoa:unit``: The `IVOA unit <https://ivoa.net/documents/VOUnits/>`__ of this column.
+:``tap:column_index``: The index of this column in a TAP table. This is used to order the columns in the table [2]_.
+:``tap:principal``: A flag indicating whether this column is the principal column in a TAP table. This should be encoded as 0 or 1 [2]_.
+:``tap:std``: A flag indicating whether this column is a standard column in a TAP table. This should be encoded as 0 or 1 [2]_.
```

Comment:
... whether this column is a representation of an element of an IVOA-standard data model.